calyx/src/lang/pretty_print.rs

```diff
@@ -171,49 +168,51 @@ impl PrettyPrint for Structure {
 
 impl PrettyPrint for Decl {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = colors::keyword(RcDoc::text("new"))
+        RcDoc::text("new")
+            .keyword()
             .append(RcDoc::space())
-            .append(colors::ident(self.name.prettify(&arena)))
+            .append(self.name.prettify(&arena).ident())
             .append(RcDoc::space())
-            .append(self.component.prettify(&arena));
-        brackets(inner)
+            .append(self.component.prettify(&arena))
+            .brackets()
     }
 }
 
 impl PrettyPrint for Std {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = colors::keyword(RcDoc::text("new-std"))
+        RcDoc::text("new-std")
+            .keyword()
             .append(RcDoc::space())
-            .append(colors::ident(self.name.prettify(&arena)))
+            .append(self.name.prettify(&arena).ident())
             .append(RcDoc::space())
             .append(self.instance.prettify(&arena))
-            .group();
-        brackets(inner)
+            .group()
+            .brackets()
     }
 }
 
 impl PrettyPrint for Wire {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = RcDoc::text("->")
+        RcDoc::text("->")
+            .keyword()
             .append(RcDoc::space())
             .append(self.src.prettify(&arena))
             .append(RcDoc::space())
-            .append(self.dest.prettify(&arena));
-        brackets(inner)
+            .append(self.dest.prettify(&arena))
+            .brackets()
     }
 }
 
 impl PrettyPrint for Compinst {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
         if self.params.is_empty() {
-            parens(self.name.prettify(&arena))
+            self.name.prettify(&arena).parens()
```

Comment:
Moving `parens` to the end seems odd. Is there a way to write `Pretty::parens(obj)`?

Reply:
yeah, but then there is no advantage of doing it this way. The whole point of doing this is to allow method chaining. idk if this better though. It's certainly a little less intuitive about what exactly `.parens()` is doing. This seems to be the style of the library though.